crypto/http/http_client.c

```diff
@@ -295,18 +297,25 @@ static int set1_content(OSSL_HTTP_REQ_CTX *rctx,
      * BIO_CTRL_INFO yields the data length at least for memory BIOs, but for
      * file-based BIOs it gives the current position, which is not what we need.
      */
-    if (BIO_get_fp(req, &fp) == 1) {
-        fseek(fp, 0, SEEK_END);
-        req_len = ftell(fp);
-        fseek(fp, 0, SEEK_SET);
+    if (BIO_method_type(req) == BIO_TYPE_FILE) {
+#ifndef OPENSSL_NO_STDIO
+        if (BIO_get_fp(req, &fp) == 1 && fseek(fp, 0, SEEK_END) == 0) {
+            req_len = ftell(fp);
+            (void)fseek(fp, 0, SEEK_SET);
+        }
+#endif
     } else {
         req_len = BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL);
         /*
          * Streaming BIOs likely will not support querying the size at all,
          * and we assume we got a correct value if req_len > 0.
          */
     }
-    if ((fp != NULL /* definitely correct req_len */ || req_len > 0)
+    if ((
+#ifndef OPENSSL_NO_STDIO
+         fp != NULL /* definitely correct req_len */ ||
```

Comment:
Now, thinking more about this - is this condition right? If the fseek() fails it means the req_len probably won't be correct. So should the condition be just `if (req_len > 0)` ?

Reply:
My aim here is to support, as far as possible, also the case that the input actually has 0 length, which usually can be checked correctly for files.
I've just added `fp = NULL` on failure, such that if `fp == NULL` the value of `req_len` is reliable even if `== 0`.